### ISSUE TITLE
Question of function run_balanced

### ISSUE THREAD

**Issue by Antinomy20001** [filaPro/visda2019#1]
https://github.com/filaPro/visda2019/blob/cb3352db3b1ff0c2f74dd6a460a65e73b259a9ca/models/common.py#L68-L81

Hi, thanks for the wonderful work! After reading the code, I have a question about the function run_balanced.

It seems to find the transposition of tensors first, and then convert it to the original order after feeding it to the models. I am confused about its function and usage...

**Comment by filaPro** [comment]
Hi @Antinomy20001,

Yes this function is a little bit tricky. The thing is that during one training step we need to [pass](https://github.com/filaPro/visda2019/blob/cb3352db3b1ff0c2f74dd6a460a65e73b259a9ca/models/mix_match.py#L34) 1 source and 2 target batches throw the same model. The sum of these 3 batch sizes doesn't allow us to pass them throw the model simultaneously due to the GPU memory limit. So we call the model 3 times. But here we are going to have troubles with `BatchNorm` statistics as they are not the same for source and target domains. This makes training process slower and much more unstable. The solution here is to form 3 batches with 1/3 of source images and 2/3 of target images in each of them.

Also, as I remember it is not our novelty, and we follow one of the open sourced `pytorch` implementations.

**Comment by Antinomy20001** [comment]
> Hi @Antinomy20001,
> 
> Yes this function is a little bit tricky. The thing is that during one training step we need to [pass](https://github.com/filaPro/visda2019/blob/cb3352db3b1ff0c2f74dd6a460a65e73b259a9ca/models/mix_match.py#L34) 1 source and 2 target batches throw the same model. The sum of these 3 batch sizes doesn't allow us to pass them throw the model simultaneously due to the GPU memory limit. So we call the model 3 times. But here we are going to have troubles with `BatchNorm` statistics as they are not the same for source and target domains. This makes training process slower and much more unstable. The solution here is to form 3 batches with 1/3 of source images and 2/3 of target images in each of them.
> 
> Also, as I remember it is not our novelty, and we follow one of the open sourced `pytorch` implementations.

Thanks for your rapid response! In fact, I am migrating your code to ```Pytorch````, hahaha...😜. 

Is this transpose operation to make ```BatchNorm```'s running_mean and running_var more stable? So call the function  ```balanced```? I think I have got it.

**Comment by filaPro** [comment]
Yes, you are right.

**Comment by Antinomy20001** [comment]
Thanks again for your response and help
